### PR TITLE
feat: add InputInt field

### PR DESCRIPTION
**Pull request by MaxPostema**

### CODE REVIEW

File: apps/molgenis-components/src/components/forms/InputInt.vue [hunk 1]
```diff
@@ -0,0 +1,67 @@
+<template>
+  <FormGroup :id="id" :label="label" :description="description">
+    <input
+      type="number"
+      step="1"
+      :value="value"
+      :class="{ 'form-control': true }"
```
Comment (connoratrug) [inline apps/molgenis-components/src/components/forms/InputInt.vue:7]: ?? why not ```class="form-control"```

File: apps/molgenis-components/src/components/forms/InputInt.vue [hunk 2]
```diff
@@ -0,0 +1,67 @@
+<template>
+  <FormGroup :id="id" :label="label" :description="description">
+    <input
+      type="number"
```
[the remark this answers is not in the document]
Comment (connoratrug) [reply apps/molgenis-components/src/components/forms/InputInt.vue:4]: how vue has no template inheritance that i know of

File: apps/molgenis-components/src/components/forms/InputInt.vue [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    emitIfValid(event) {
+      const value = parseInt(event.target.value);
+      if (event.target.value === "") {
+        this.$emit("input", null);
+      }
+      if (!isNaN(value)) {
+        this.$emit("input", value);
+      }
+    },
+    handleKeyValidity(event) {
```
Comment (mswertz) [inline apps/molgenis-components/src/components/forms/InputInt.vue:36]: I realize this is a new feature request so you can ignore. However, LongInput has validation on wether it is a big int. Int doesn't. I propose we more or less merge LongInput and IntInput with only difference the min and max?

**Review by mswertz — CHANGES_REQUESTED** [review]

File: apps/molgenis-components/src/components/forms/InputInt.vue (hunk 3 above)
Comment (chinook25) [reply apps/molgenis-components/src/components/forms/InputInt.vue:36]: One big difference is that Long is actually a string while int returns a number. So if we want to merge them we'd have to think about how to treat them. InputInt is a number in order to use it in expressions.

Comment (mswertz) [reply apps/molgenis-components/src/components/forms/InputInt.vue:36]: ah, then ignore my 'merge' comment.
My only comment remaining is that the 'Int' will not check if it goes beyond what is allowed. Server will complain though. So you decide if you want to make the change or postpone for later.

**Review by mswertz — APPROVED** [review]

**Review by connoratrug — APPROVED** [review]

**Pull request merged**